sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -159,7 +159,7 @@ void commit() {
         private BlockBlobOutputStream(final BlobAsyncClient client,
             final ParallelTransferOptions parallelTransferOptions, final BlobHttpHeaders headers,
             final Map<String, String> metadata, final AccessTier tier, final BlobRequestConditions requestConditions) {
-            super(BlockBlobClient.MAX_STAGE_BLOCK_BYTES);
+            super(Integer.MAX_VALUE);
```

Comment:
It ends up being used here https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageOutputStream.java#L54 .

As far as I can tell from reading inheritance tree of StorageOutputStream - this can't be changed to long because of conversion to Flux of ByteBuffers - these are constrained by int capacity - these conversions happen in subtypes. 

@gapra-msft @rickle-msft  do you guys have any context behind previously chosen value?

I think this could be set to `Math.min(Integer.MAX_VALUE, parallelTransferOptions.getBlockSizeLong())` so that it would result in ByteBuffers already trimmed nice for staging blocks (for small blocks) and largest possible ByteBuffers for jumbo blocks.

Reply:
Actually, specifically for BlockBlobOutputStream, after my changes to make it write via buffered upload, this this.writeThreshold isn't used. If you look at BlockBlobOutputStream, writeInternal is overridden and dispatchWrite is never called (it just returns an empty mono). Could you just add a note about that if you don't mind?

Reply:
Basically all the write Threshold logic is handled within buffered upload

Reply:
added comment per our discussion.